### DISCUSSION TITLE
Zeek configured correctly, but no logs appear

### DISCUSSION THREAD

**Discussion by pktmterenceg** [Security-Onion-Solutions/securityonion#2779] [Unsupported Versions]
Hi all!

Thanks to everyone involved in this community! I appreciate the ability to ask questions and get help. 

I've got something here which I don't see discussed elsewhere (forgive me, it might be difficult search keys) which I'd like to ask the community about, please. 

I've got SO installed on an old Mac mini, with upgraded RAM, etc. The installation went reasonably well (`so-status` shows all `OK`s), and everything /seems/ to be running. 

However - and this is weird - Zeek doesn't appear to be producing any logs whatsoever. By this, I mean that `/nsm/zeek/logs` appears to have started logging data (on the 21st) and has logged nothing since then. 

I've confirmed that data is flowing to the SO box (by doing `tcpdump` on `bond0`). I've confirmed free disk space (`df -h` shows 327G free on` /`). I started and stopped the service (`so-status` still giving `OK`), and didn't get any errors in messages or syslog. 

But I'm now out of ideas. Anyone else seen anything like this? Are there other places I can look for diagnostic data?

Thanks much!

**Comment by innovate-support** [comment]
Is that bare metal installation of SO, or is it a VM?

I'm not sure this is your issue or not, but it sounds like your monitoring input is just dark.  Is your monitoring interface receiving data from a Tap? or from a switch with a Mirrored port?  Or if a VM it needs to have Promiscuous mode enabled with traffic flashing on that interface.

**Reply by pktmterenceg** [reply]
Hi,

Thanks for the reply!

Yes, bare metal. And, yes, I've confirmed I'm getting input from the router's port mirroring feature. As I meant to say above, I've sniffed (`tcpdump`) both the physical interface, and the `bond0` interface, and got expected data from each. 

See why I'm out of ideas? :)

**Comment by TOoSmOotH** [comment]
Are you getting any alerts from Suricata? The logs for zeek are in /nsm/zeek would be a place to start. What install mode? "sudo docker logs so-zeek" is another things as well.

**Reply by pktmterenceg** [reply]
Hey Toosmooth, 

Thanks for the reply/suggestions. 

I am getting alerts from Suricata, and those alerts appear to be recent/continuing. I can't say if the volume is appropriate, of course. 

The output of your suggested command, though, is somewhat interesting, and I suspect a good lead on where to look next, as you'll see here:

`sudo docker logs so-zeek
[sudo] password for administrator: 
Error: failed to write file: [Errno 13] Permission denied: '/nsm/zeek/spool/.zeekctl-config.sh.tmp'
Error: cannot acquire lock: [Errno 13] Permission denied: '/nsm/zeek/spool/lock.10'
Error: Unable to get lock`

That's weird, right? 

Thank you, again!

**Comment by pktmterenceg** [comment]
Can I bump this, please? Any thoughts given the odd permissions error I posted the other day? Thanks again!!

**Reply by cm-ops** [reply]
You could try stopping Zeek, moving /nsm/zeek/spool directory to /home, then restart Zeek. The /nsm/zeek/spool directory should be recreated on restart and check if you are receiving Zeek logs.

**Reply by pktmterenceg** [reply]
Hey cm-ops!

Thanks for this suggestion! I tried this, and logs now appear to be populating! (w00t!)

Additionally, the output from `sudo docker logs so-zeek` appears healthier, too:

`sudo docker logs so-zeek
checking configurations ...
installing ...
creating policy directories ...
installing site policies ...
generating cluster-layout.zeek ...
generating local-networks.zeek ...
generating zeekctl-config.zeek ...
generating zeekctl-config.sh ...
stopping ...
stopping worker ...
stopping proxy ...
stopping manager ...
stopping logger ...
starting ...
starting logger ...
starting manager ...
starting proxy ...
starting worker ...`

I'll check again in the AM and update once Zeek has successfully rolled over to a new day. Thanks again!!

**Reply by pktmterenceg** [reply]
Quick update: THANK YOU to all who helped out with this. I'm now getting Zeek logs as one would expect. 👍
